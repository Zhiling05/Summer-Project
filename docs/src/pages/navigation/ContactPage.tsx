// src/pages/navigation/SettingsPage.tsx
export default function ContactPage() {
  return <h1>SettingsPage</h1>;
}
