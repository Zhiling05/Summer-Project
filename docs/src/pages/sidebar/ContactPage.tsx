// src/pages/sidebar/ContactPage.tsx
export default function ContactPage() {
  return <h1>Contact us page</h1>;
}
