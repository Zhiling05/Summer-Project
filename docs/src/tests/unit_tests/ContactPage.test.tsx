import { render, screen } from '@testing-library/react';
import ContactPage from '../../pages/sidebar/ContactPage';

jest.mock('../../components/Header', () => ({ title }: { title: string }) => (
  <div data-testid="header">{title}</div>
));

describe('ContactPage Component', () => {
  test('renders all contact sections and info', () => {
    render(<ContactPage />);

    // Header
    expect(screen.getByTestId('header')).toHaveTextContent('Contact Us');
    expect(screen.getByRole('heading', { name: 'Contact Us' })).toBeInTheDocument();

    // Contact Information Section
    expect(screen.getByRole('heading', { name: 'Contact Information' })).toBeInTheDocument();
    expect(screen.getByText(/If you have any questions or need further assistance/i)).toBeInTheDocument();
    expect(screen.getByText(/http:\/\/XXXXXXXXX.com/)).toBeInTheDocument();
    expect(screen.getByText(/XXXXXXX@example.com/)).toBeInTheDocument();
    expect(screen.getByText('+1 (234) 567-890')).toBeInTheDocument();
    expect(screen.getByText(/1234 Research Lane/)).toBeInTheDocument();

    // Follow Us Section
    expect(screen.getByRole('heading', { name: 'Follow Us' })).toBeInTheDocument();
    expect(screen.getByText(/Stay updated with the latest news/i)).toBeInTheDocument();
    expect(screen.getByText('Twitter')).toBeInTheDocument();
    expect(screen.getByText('LinkedIn')).toBeInTheDocument();

    // Support Hours Section
    expect(screen.getByRole('heading', { name: 'Support Hours' })).toBeInTheDocument();
    expect(screen.getByText(/Our support team is available Monday through Friday/i)).toBeInTheDocument();
  });
});
