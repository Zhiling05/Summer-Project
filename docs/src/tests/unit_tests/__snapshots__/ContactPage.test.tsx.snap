// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    SettingsPage
  </h1>
</DocumentFragment>
`;
